File: dpo_train/generation_output/cpp-java/HumanEval_102_choose_num.java
### C++ version

#include<assert.h>
#include<bits/stdc++.h>
#include<vector>
#include<sstream>
#include<regex>
#include<map>
#include<set>
long choose_num(long x, long y) {
    if (x > y) {
        return -1;
    }
    if (y % 2 == 0) {
        return y;
    }
    if (x == y) {
        return -1;
    }
    return y - 1;
}

### Java version

import java.util.*;
import java.lang.reflect.*;
import org.javatuples.*;
import java.security.*;
import java.math.*;
import java.io.*;
import java.util.stream.*;
class Problem {
    public static long chooseNum(long x, long y) {
        return choose_num(x, y);
    }
    public static long choose_num(long x, long y) {
        if (x == y) {
            return x;
        }
        if (x % 2 == 0) {
            return x;
        }
        if (y % 2 == 0) {
            return y;
        }
        return y - 1;
    }
}

class Solution {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int n = in.nextInt();
        int m = in.nextInt();
        int[] nums = new int[n];
        for (int i = 0; i < n; i++) {
            nums[i] = in.nextInt();
        }
        long[] nums2 = new long[m];
        for (int i = 0; i < m; i++) {
            nums2[i] = in.nextInt();
        }
        long res = Problem.chooseNum(nums, nums2);
        System.out.println(res);
    }
}

File: test/before_train/trainset_and_testset/py-cpp/HumanEval_102_choose_num.cpp
#include<assert.h>
#include<bits/stdc++.h>
#include<vector>
#include<sstream>
#include<regex>
#include<map>
#include<set>
#include<unordered_map>
#include<unordered_set>
long choose_num(long x, long y) {
    if (x == y) {
        return x;
    }
    if (x > y) {
        return -1;
    }
    if (x == y) {
        return -1;
    }
    return x - 1;
}
int main() {
    auto candidate = choose_num;
    assert(candidate((12), (15)) == (14));
    assert(candidate((13), (12)) == (-1));
    assert(candidate((33), (12354)) == (12354));
    assert(candidate((5234), (5233)) == (-1));
    assert(candidate((6), (29)) == (28));
    assert(candidate((27), (10)) == (-1));
    assert(candidate((7), (7)) == (-1));
    assert(candidate((546), (546)) == (546));
}
